apps/www/components/Nav/RightClickBrandLogo.tsx

```diff
@@ -0,0 +1,281 @@
+import React, { Fragment, MouseEvent, PropsWithChildren, ReactNode, useRef, useState } from 'react'
+import Image from 'next/image'
+import Link from 'next/link'
+import { useClickAway } from 'react-use'
+import {
+  cn,
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuPortal,
+  DropdownMenuTrigger,
+  IconCheck,
+  IconClipboard,
+} from 'ui'
+
+import * as supabaseLogoWordmarkDark from 'common/assets/images/supabase-logo-wordmark--dark.png'
+import * as supabaseLogoWordmarkLight from 'common/assets/images/supabase-logo-wordmark--light.png'
+
+/**
+ * Right click on the Supabase logo in the website navbar
+ * for quick access to brand assets.
+ */
+const RightClickBrandLogo = () => {
+  const ref = useRef(null)
+  const triggerRef = useRef(null)
+  const [open, setOpen] = useState<boolean>(false)
+
+  const handleRightClick = (e: MouseEvent) => {
+    e.preventDefault()
+
+    if (e.type === 'contextmenu' || e.nativeEvent.button === 2) {
+      setOpen(true)
+    }
+  }
+
+  useClickAway(ref, () => {
+    setOpen(false)
+  })
+
+  return (
+    <DropdownMenu open={open}>
+      <DropdownMenuTrigger onContextMenu={handleRightClick} asChild>
+        <Link
+          ref={triggerRef}
+          href="/"
+          className="block w-auto h-6 focus-visible:ring-2 focus-visible:outline-none focus-visible:ring-foreground-lighter focus-visible:ring-offset-4 focus-visible:ring-offset-background-alternative focus-visible:rounded-sm"
+        >
+          <Image
+            src={supabaseLogoWordmarkLight}
+            width={124}
+            height={24}
+            alt="Supabase Logo"
+            className="dark:hidden"
+            priority
+          />
+          <Image
+            src={supabaseLogoWordmarkDark}
+            width={124}
+            height={24}
+            alt="Supabase Logo"
+            className="hidden dark:block"
+            priority
+          />
+        </Link>
+      </DropdownMenuTrigger>
+      <DropdownMenuPortal container={triggerRef.current}>
+        <DropdownMenuContent ref={ref} align="start" side="bottom" className="mt-2 p-0 w-52">
+          {menuItems.map((section, sectionIdx) => (
+            <Fragment key={`cxtMenu-section-${sectionIdx}`}>
+              {sectionIdx !== 0 && <Divider />}
+              <div className="p-1">
+                {section.map((menuItem) => (
+                  <DropdownMenuItem
+                    key={menuItem.label}
+                    className="group/menu-item w-full flex justify-between gap-2 items-center p-2"
+                  >
+                    <MenuItem
```

Comment:
Confusing name for a local component, as there is a MenuItem component in www/components

Reply:
updated